File: test-junit5/src/test/java/io/micronaut/test/junit5/DisableEnvironmentDeductionTest.java
package io.micronaut.test.junit5;

import io.micronaut.context.env.Environment;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import jakarta.inject.Inject;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest(deduceEnvironment = false)
class DisableEnvironmentDeductionTest {

    @Inject
    Environment environment;

    @Test
    void environmentDeductionPropertyIsFalse() {
        assertEquals(Boolean.FALSE, environment.getProperty(Environment.DEDUCE_ENVIRONMENT_PROPERTY, Boolean.class).orElse(null));
    }
}

File: test-kotest5/src/test/kotlin/io/micronaut/test/kotest5/DisableEnvironmentDeductionTest.kt
package io.micronaut.test.kotest5

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.env.Environment
import io.micronaut.test.extensions.kotest5.annotation.MicronautTest
import jakarta.inject.Inject

@MicronautTest(deduceEnvironment = false)
internal class DisableEnvironmentDeductionTest : StringSpec() {

    @Inject
    lateinit var environment: Environment

    init {
        "environment deduction property is false" {
            environment.getProperty(Environment.DEDUCE_ENVIRONMENT_PROPERTY, Boolean::class.java).orElse(null) shouldBe false
        }
    }
}
